crates/c-api/include/wasmtime.h

```diff
@@ -67,6 +68,39 @@ WASM_API_EXTERN bool wasmtime_wat2wasm(
     own wasm_byte_vec_t *error_message
 );
 
+#define WASMTIME_DECLARE_OWN(name) \
+  typedef struct wasmtime_##name##_t wasmtime_##name##_t; \
+  \
+  WASM_API_EXTERN void wasmtime_##name##_delete(own wasmtime_##name##_t*);
+
+WASMTIME_DECLARE_OWN(linker)
+
+WASM_API_EXTERN wasmtime_linker_t* wasmtime_linker_new(wasm_store_t* store);
+
+WASM_API_EXTERN bool wasmtime_linker_define(
+    wasmtime_linker_t *linker,
+    const wasm_name_t *module,
+    const wasm_name_t *name,
+    const wasm_extern_t *item
+);
+
+WASM_API_EXTERN bool wasmtime_linker_define_wasi(
+    wasmtime_linker_t *linker,
+    wasi_instance_t *instance
+);
+
+WASM_API_EXTERN bool wasmtime_linker_define_instance(
+    wasmtime_linker_t *linker,
+    const wasm_name_t *name,
+    wasm_instance_t *instance
+);
+
+WASM_API_EXTERN wasm_instance_t* wasmtime_linker_instantiate(
+    wasmtime_linker_t *linker,
```

Comment:
Assuming `Linker::instantiate` can become `&self`, this (and the corresponding Rust impl) should probably be `const wasmtime_linker_t *`.